### PR TITLE
Transparent statusbar

### LINKED ISSUE #493: Make status bar transparent when on details of a picture so that the picture can be seen through the status bar. Same when previewing a picture.
See Extensions.kt:217 for some first attempt at solving this

### PR DESCRIPTION
**Pull request by LunarX**
Closes #493

### CODE REVIEW

File: app/src/main/java/com/infomaniak/drive/ui/fileList/fileDetails/FileDetailsFragment.kt [hunk 1]
```diff
@@ -54,6 +57,18 @@ class FileDetailsFragment : FileDetailsSubFragment() {
 
         toolbar.setNavigationOnClickListener { findNavController().popBackStack() }
 
+        // If light mode, change the color of the icons of the status to match the background
+        // If dark mode the icons stay white all along, no need to check
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/drive/ui/fileList/fileDetails/FileDetailsFragment.kt:61]:
```suggestion
        // If in Light mode, change the status icons color to match the background.
        // If in Dark mode, the icons stay white all along, no need to check.
```

File: app/src/main/java/com/infomaniak/drive/ui/fileList/preview/PreviewSliderFragment.kt [hunk 2]
```diff
@@ -57,6 +57,7 @@ import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
 
+
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/drive/ui/fileList/preview/PreviewSliderFragment.kt:60]:
```suggestion
```

File: app/src/main/java/com/infomaniak/drive/ui/MainActivity.kt [hunk 3]
```diff
@@ -298,6 +299,14 @@ class MainActivity : BaseActivity() {
 
         handleBottomNavigationVisibility(destination.id, shouldHideBottomNavigation)
 
+        if (destination.id != R.id.previewSliderFragment && destination.id != R.id.fileDetailsFragment) {
+            if (lastDestinationId == R.id.previewSliderFragment || lastDestinationId == R.id.fileDetailsFragment) {
+                window.toggleEdgeToEdge(false)
+            }
+            bottomNavigation.setOnApplyWindowInsetsListener(null)
+        }
+        lastDestinationId = destination.id
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/drive/ui/MainActivity.kt:308]: Isn't there a better way ?

File: app/src/main/java/com/infomaniak/drive/ui/fileList/preview/PreviewSliderFragment.kt [hunk 4]
```diff
@@ -198,8 +255,9 @@ class PreviewSliderFragment : Fragment(), FileInfoActionsView.OnItemClickListene
                 override fun onStateChanged(bottomSheet: View, newState: Int) {
                     when (bottomSheetBehavior.state) {
                         BottomSheetBehavior.STATE_HIDDEN -> {
-                            activity?.window?.navigationBarColor =
-                                ContextCompat.getColor(requireContext(), R.color.previewBackground)
+                            activity?.window?.navigationBarColor = ColorUtils.setAlphaComponent(
+                                ContextCompat.getColor(requireContext(), R.color.previewBackground), 128
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/drive/ui/fileList/preview/PreviewSliderFragment.kt:259]: Fix magic value

File: app/src/main/java/com/infomaniak/drive/ui/fileList/preview/PreviewSliderFragment.kt [hunk 5]
```diff
@@ -213,39 +271,39 @@ class PreviewSliderFragment : Fragment(), FileInfoActionsView.OnItemClickListene
...
     }
 
-    override fun onResume() {
-        super.onResume()
-
+    private fun setupTransparentStatusBar() {
         activity?.window?.apply {
-            statusBarColor = ContextCompat.getColor(requireContext(), R.color.previewBackground)
+            statusBarColor = ColorUtils.setAlphaComponent(
+                ContextCompat.getColor(requireContext(), R.color.previewBackground), 128
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/drive/ui/fileList/preview/PreviewSliderFragment.kt:277]: Same, magic value

File: app/src/main/java/com/infomaniak/drive/ui/fileList/preview/PreviewSliderFragment.kt [hunk 6]
```diff
@@ -213,39 +269,37 @@ class PreviewSliderFragment : Fragment(), FileInfoActionsView.OnItemClickListene
...
-            observeOfflineProgression(this@PreviewSliderFragment) { fileId ->
-                previewSliderAdapter.updateFile(fileId) { file -> file.isOffline = true }
-            }
-        }
-    }
+        view?.apply {
+            setOnApplyWindowInsetsListener(null) // This fixes an issue where the material bottom navigation's icons disappear
+            ViewCompat.setOnApplyWindowInsetsListener(this) { _, windowInsets ->
+                with(windowInsets.getInsets(WindowInsetsCompat.Type.systemBars())) {
+                    val peekHeight = getDefaultPeekHeight()
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/drive/ui/fileList/preview/PreviewSliderFragment.kt:284]: Use once, so no need for a variable

**Review by sirambd — CHANGES_REQUESTED** [review]
Wait changes

**Review by sirambd — APPROVED** [review]

**Review by JorisBodin — CHANGES_REQUESTED** [review]
Problem with CollapsingSubTitleToolbarBehavior in FileDetailsFragment

**Review by JorisBodin — APPROVED** [review]
https://github.com/Infomaniak/android-kDrive/issues/622

**Pull request merged**